### PR TITLE
[FIX] - Fix CSV Output from Export Logs

### LINKED ISSUE #110: Fix CSV Output from Export Logs
In the CSV that's created, there's columns that don't need to be present. For example, these columns can be removed:
- `attnTo`- the user id of the employee that's marked as "attention to" is irrelevant to the user, since the first and last names of the employee are present
- `employeeId`- same reason as above
- `logId`- not needed for the user

Some other changes we can look into, but not required:
- [x] Combine the first name and last name of attnTo and employee without changing the route
- [x] Naming the file after when it was exported (i.e "log_records 6-19-23")

### PR DESCRIPTION
**Pull request by owen-sellner**
## GitHub Issue link

[Fix CSV Output from Export Logs](https://github.com/uwblueprint/supportive-housing/issues/110)

## Implementation description
* Created `CSVLog` type to remove the properties logId and employeeId exported into the csv
* Combined the first name and last name properties for attnTo and employees 
* Added function `convertToCSVLog` to convert from `LogRecord` to `CSVLog`
* Added the current date to the file name for the CSV file

## Steps to test
1. Verify the export button creates and downloads a CSV file
2. Verify the CSV file contains all of the log records within the date range given
3. Verify the data in the CSV correctly matches the heading it is under

## What should reviewers focus on?
* Can the headers for the CSV file be updated automatically if the properties of `CSVLog` are changed?
* Should the first name and last name of residents be combined?
* What format should the date in the name of the CSV file be? (Currently it is YYYY-MM-DD)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [x] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [x] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [x] I have updated other Docs as needed

### CODE REVIEW

File: frontend/src/types/CSVLog.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export type CSVLog = {
+    attnTo: string;
+    building: string;
+    datetime: string;
+    employee: string;
+    flagged: boolean;
+    note: string;
+    residentFirstName: string;
```
Comment (safchow) [inline frontend/src/types/CSVLog.ts]: we actually don't store the resident's name in a log record, so you can update this to `residentID` (also remove lastName)

File: frontend/src/helper/CSVConverter.tsx [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
+  };
+};
 
 const CSVConverter = (data: LogRecord[]): boolean => {
   // Convert JSON to CSV
   try {
     const csvRows = [];
 
-    const headers = Object.keys(data[0]);
+    const headers = ["attnTo", "building", "datetime", "employee", "flagged", "note", "residentFirstName", "residentLastName", "tags"];
```
Comment (safchow) [inline frontend/src/helper/CSVConverter.tsx]: should be:
```suggestion
    const headers = ["attnTo", "building", "datetime", "employee", "flagged", "note", "residentID", "tags"];
```

File: frontend/src/helper/CSVConverter.tsx [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
+  
+  return {
+    attnTo,
+    building: logRecord.building,
+    datetime: logRecord.datetime,
+    employee,
+    flagged: logRecord.flagged,
+    note: logRecord.note,
+    residentFirstName: logRecord.residentFirstName,
+    residentLastName: logRecord.residentLastName,
```
Comment (safchow) [inline frontend/src/helper/CSVConverter.tsx]: change this to id:
```suggestion
    residentId: logRecord.residentId
```

Comment (safchow) [reply frontend/src/helper/CSVConverter.tsx]: and also remove first name

**Review by safchow — CHANGES_REQUESTED** [review]
thanks for getting this fix out super quickly :D 

There are a few suggestions for how residents are handled, but otherwise, all the asks in the ticket were met :rocket:

File: frontend/src/components/pages/HomePage/LogRecordsTable.tsx [hunk 4]
```diff
@@ -60,7 +60,7 @@ const LogRecordsTable = ({
                   {
                     // TODO: Resolve the resident record at some point
```
Comment (Safewaan) [inline frontend/src/components/pages/HomePage/LogRecordsTable.tsx:61]: you can remove this todo

**Review by safchow — APPROVED** [review]
lgtm (looks garbage to me)

File: frontend/src/types/CSVLog.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export type CSVLog = {
+    attnTo: string;
+    building: string;
+    datetime: string;
+    employee: string;
+    flagged: boolean;
+    note: string;
+    residentId: string;
+    tags: string;
+};
```
Comment (kevin-pierce) [inline frontend/src/types/CSVLog.ts]: nit: I think we might want to run the linter on this 🤓🤓 (we use default ESLint settings, and I think it enforces empty lines in files)

File: frontend/src/helper/CSVConverter.tsx [hunk 6]
```diff
@@ -20,7 +36,11 @@ const CSVConverter = (data: LogRecord[]): boolean => {
     const url = URL.createObjectURL(blob);
     const link = document.createElement("a");
     link.href = url;
-    link.setAttribute("download", "log_records.csv");
+    
+    const dateToday = new Date();
+    const dateTodayString = dateToday.toLocaleString("fr-CA", {timeZone: "America/Toronto"}).substring(0, 10);
```
Comment (kevin-pierce) [inline frontend/src/helper/CSVConverter.tsx]: nit: Maybe add a comment above explaining why we're leveraging the `fr-CA` formatter 

For instance,

```suggestion
    // Format the date into YYYY-MM-DD 
    const dateToday = new Date();
    const dateTodayString = dateToday.toLocaleString("fr-CA", {timeZone: "America/Toronto"}).substring(0, 10);
```

**Review by kevin-pierce — APPROVED** [review]
LGTM! The nits are super minor, feel free to ignore them

**Pull request merged**